app/webpacker/components/Live/Admin/Results/index.jsx

```diff
@@ -0,0 +1,200 @@
+  const [error, setError] = useState('');
+  const [success, setSuccess] = useState('');
+  const queryClient = useQueryClient();
+
+  const { data: results, isLoading } = useQuery({
+    queryKey: [roundId, 'results'],
+    queryFn: () => getRoundResults(roundId, competitionId),
+  });
+
+  const handleRegistrationIdChange = useCallback((_, { value }) => {
+    setRegistrationId(value);
+    const alreadyEnteredResults = results.find((r) => r.registration_id === value);
+    if (alreadyEnteredResults) {
+      setAttempts(alreadyEnteredResults.attempts.map((a) => a.result));
+    } else {
+      setAttempts(zeroedArrayOfSize(solveCount));
+    }
+  }, [results, solveCount]);
+
+  const {
+    mutate: mutateSubmit, isPending: isPendingSubmit,
+  } = useMutation({
+    mutationFn: submitRoundResults,
+    onSuccess: () => {
+      setSuccess('Results added successfully!');
+      setRegistrationId('');
+      setAttempts(zeroedArrayOfSize(solveCount));
+      setError('');
+
+      setTimeout(() => setSuccess(''), 3000);
+    },
+    onError: () => {
+      setError('Failed to submit results. Please try again.');
+    },
+  });
+
+  const {
+    mutate: mutateUpdate, isPending: isPendingUpdate,
+  } = useMutation({
+    mutationFn: updateRoundResults,
+    onSuccess: () => {
+      setSuccess('Results added successfully!');
+      setRegistrationId('');
+      setAttempts(zeroedArrayOfSize(solveCount));
+      setError('');
+
+      setTimeout(() => setSuccess(''), 3000);
+    },
+    onError: () => {
+      setError('Failed to submit results. Please try again.');
+    },
+  });
+
+  useEffect(() => {
+    const cable = createConsumer();
+
+    const subscription = cable.subscriptions.create(
+      { channel: 'LiveResultsChannel', round_id: roundId },
+      {
+        received: (data) => {
+          queryClient.setQueryData([roundId, 'results'], (oldData) => {
+            const existingIndex = oldData.map((a) => a.registration_id).indexOf(data.registration_id);
+            if (existingIndex === -1) {
+              return [...oldData, data];
+            }
+            return oldData.map((a) => (a.registration_id === data.registration_id ? data : a));
+          });
+        },
+      },
+    );
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [roundId, queryClient, eventId]);
+
+  const handleAttemptChange = (index, value) => {
+    const newAttempts = [...attempts];
+    newAttempts[index] = value;
+    setAttempts(newAttempts);
+  };
+
+  const handleSubmit = async () => {
+    if (!registrationId) {
+      setError('Please enter a user ID');
+      return;
+    }
+
+    if (results.find((r) => r.registration_id === registrationId)) {
+      mutateUpdate({
+        roundId, registrationId, competitionId, attempts,
+      });
+    } else {
+      mutateSubmit({
+        roundId, registrationId, competitionId, attempts,
+      });
+    }
+  };
+
+  return (
```

Comment:
Just from the fact that I wrote a lot of "same as above" comments below: How much code is this actually sharing with `DoubleCheck`?

Reply:
this definitely can be combined